presto-main/src/main/java/com/facebook/presto/type/TypeUtils.java

```diff
@@ -82,7 +82,7 @@ else if (type.getJavaType() == Slice.class) {
                 return (long) methodHandle.invoke(type.getSlice(block, position));
             }
             else {
```

Comment:
Make this an `else if (!type.getJavaType().isPrimitive())` and keep the `else` that throws an exception. That way when we add support for `int.class` this will still fail with the right message